Lib/asyncio/tasks.py

```diff
@@ -745,7 +745,10 @@ async def _wrap_awaitable(awaitable):
     if loop is None:
         loop = events.get_event_loop()
     try:
-        return loop.create_task(coro_or_future)
+        if context is None:
+            return loop.create_task(coro_or_future)
+        else:
+            return loop.create_task(coro_or_future, context=context)
```

Comment:
If I read the [definition of `create_task`](https://github.com/python/cpython/blob/95f5c89b545beaafad73f05a695742da3e90bc41/Lib/asyncio/tasks.py#L402) correctly, I think we can pass in `context` directly, since we also do a `None` check in `create_task` (unless I missed something).

Reply:
Removing the if/else here causes a couple of test failures:

2 tests failed:
    test.test_asyncio.test_base_events test_ssl

The version of create_task being called here is actually the one in base_events.py (line 462), and the problem is that the current unit tests define their own EventLoop class which supports only the old API, without a context argument. For example, in test_base_events, there's the following:

```python
    def test_create_task(self):
        class MyTask(asyncio.Task):
            pass

        async def test():
            pass

        class EventLoop(base_events.BaseEventLoop):
            def create_task(self, coro):
                return MyTask(coro, loop=loop)

        loop = EventLoop()
        self.set_event_loop(loop)

        coro = test()
        task = asyncio.ensure_future(coro, loop=loop)
        self.assertIsInstance(task, MyTask)

        # make warnings quiet
        task._log_destroy_pending = False
        coro.close()
```

This test could be updated to support the new API, but this exact issue could also appear in application code which defines its own event loop based on the old API. So, I think the safest thing to do is to only call the new API when a context is actually passed in.

Reply:
Make sense, thanks for clarifying!